src/bin/elasticsearch_exporter.rs

```diff
@@ -0,0 +1,359 @@
+#![feature(str_split_once)]
+
+#[macro_use]
+extern crate lazy_static;
+#[macro_use]
+extern crate prometheus;
+#[macro_use]
+extern crate log;
+
+use clap::Clap;
+use hyper::server::conn::AddrStream;
+use hyper::{
+    header::CONTENT_TYPE,
+    service::{make_service_fn, service_fn},
+    Body, Request, Response, Server, StatusCode,
+};
+use prometheus::{Encoder, HistogramVec, TextEncoder, TEXT_FORMAT};
+use std::convert::Infallible;
+use std::env;
+use std::error::Error as StdError;
+use std::fmt;
+use std::net::SocketAddr;
+use std::panic;
+use std::str::FromStr;
+use std::time::Duration;
+use tokio::signal;
+use tokio::sync::oneshot::{self, Receiver, Sender};
+use url::Url;
+
+use elasticsearch_exporter::{
+    CollectionLabels, Exporter, ExporterOptions, ExporterPollIntervals, Labels,
+};
+
+lazy_static! {
+    static ref HTTP_REQ_HISTOGRAM: HistogramVec = register_histogram_vec!(
+        "http_request_duration_seconds",
```

Comment:
Would it make sense to go down to milliseconds for better detail?

Reply:
Initially I went with milliseconds, but his is something prometheus recommends https://prometheus.io/docs/practices/naming.
Also histogram bucket allows to modify this granularity into milliseconds.